### PR TITLE
Generate the CHANGELOG based on the PR's target branch [skip ci]

### PR DESCRIPTION
**Pull request by NvTimLiu**
To fix: https://github.com/NVIDIA/spark-rapids/issues/11866

Generate the CHANGELOG based on the PR's target branch if the PR's project roadmap is empty.

### CODE REVIEW

**Review by YanxuanLiu — COMMENTED** [review]
Do we still need `no_project_prs` after your changes? Seems it can be removed.

**Comment by NvTimLiu** [comment]
> no_project_prs
updated, remove the unused varaible

**Review by YanxuanLiu — APPROVED** [review]
LGTM

**Review by pxLi — DISMISSED** [review]
LGTM. assume this has been verified with different cases thanks

File: scripts/generate-changelog [hunk 1]
```diff
@@ -183,9 +182,13 @@ def process_changelog(resource_type: str, changelog: dict, releases: set, projec
...
                         continue  # skip auto-gen PR
-                    no_project_prs.append(item)
-                continue
-            project = item['projectCards']['nodes'][0]['project']['name']
+                    # Obtain the version from the PR's target branch, e.g. branch-x.y --> x.y
+                    ver = item['baseRefName'].replace('branch-', '')
+                    project = f"{RELEASE} {ver}"
+                else:
+                    continue
+            else:
```
Comment (pxLi) [inline scripts/generate-changelog:190]: nit: this should be safe to be removed, as it was trying make the script compatible for classic projects which was gone for a long time. its fine we could still leave it here

**Comment by NvTimLiu** [comment]
> LGTM. assume this has been verified with different cases thanks

Yes verified PRs/Issues both with and without project[roadmap], works well

**Comment by NvTimLiu** [comment]
Build PASS:

```
$ scripts/generate-changelog  --releases=24.10,24.12
  Generating changelog ...
  Processing pull requests ...
  Processing issues ...
  Done.
```

File: scripts/generate-changelog (hunk 1 above)
Comment (NvTimLiu) [reply scripts/generate-changelog:186]: Agree; removing compatibility with project cards simplifies the logic."

**Review by pxLi — APPROVED** [review]

**Comment by NvTimLiu** [comment]
build

**Pull request merged**